pandas/io/formats/style.py

```diff
@@ -340,6 +343,319 @@ def to_excel(
+        multicol_align : {"r", "c", "l"}
+            If sparsifying hierarchical MultiIndex columns whether to align text at
+            the left, centrally, or at the right.
+        siunitx : bool, default False
+            Set to ``True`` to structure LaTeX compatible with the {siunitx} package.
+        encoding : str, default "utf-8"
+            Character encoding setting.
+
+        Returns
+        -------
+        str or None
+            If `buf` is None, returns the result as a string. Otherwise returns `None`.
+
+        See Also
+        --------
+        Styler.format: Format the text display value of cells.
+
+        Notes
+        -----
+        **Latex Packages**
+
+        For the following features we recommend the following LaTeX inclusions:
+
+        ===================== ==========================================================
+        Feature               Inclusion
+        ===================== ==========================================================
+        sparse columns        none: included within default {tabular} environment
+        sparse rows           \\usepackage{multirow}
+        hrules                \\usepackage{booktabs}
+        colors                \\usepackage[table]{xcolor}
+        siunitx §             \\usepackage{siunitx}
+        bold (with siunitx)   | \\usepackage{etoolbox}
+                              | \\robustify\\bfseries
+                              | \\sisetup{detect-all = true}  *(within {document})*
+        italic (with siunitx) | \\usepackage{etoolbox}
+                              | \\robustify\\itshape
+                              | \\sisetup{detect-all = true}  *(within {document})*
+        ===================== ==========================================================
+
+        §: if using the ``siunitx`` argument then numeric columns will be set to a
+        ``column_format`` of "S" instead of "r".
+
+        **Cell Styles**
+
+        LaTeX styling can only be rendered if the accompanying styling functions have
+        been constructed with appropriate LaTeX commands. All styling
+        functionality is built around the concept of a CSS ``(<attribute>, <value>)``
+        pair (see `Table Visualization <../../user_guide/style.ipynb>`_), and this
+        should be replaced by a LaTeX
+        ``(<command>, <options>)`` approach and each cell will be styled individually
+        using nested LaTeX commands with their accompanied options.
+
+        For example the following code will highlight and bold a cell in HTML-CSS:
+
+        >>> df = pd.DataFrame([[1,2], [3,4]])
+        >>> s = df.style.highlight_max(axis=None,
+        ...                            props='background-color:red; font-weight:bold;')
+        >>> s.render()
+
+        The equivalent using LaTeX only commands is the following:
+
+        >>> s = df.style.highlight_max(axis=None,
+        ...                            props='cellcolor:{red}; bfseries: ;')
```

Comment:
No, not unless we change how to parse CSS - a ':' is always present, and raises currently if missing.

Reply:
A colon plus empty value plays more nicely with the additional wrapping options below, I think.

Reply:
Right - I was thinking of parsing before we get to that step, something like:

```
props = 'cellcolor:{red}; bfseries;'
props = ';'.join([e + ':' if ':' not in e else e for e in props.split(';')])
props = props[:-1] if props[-1] == ':' else props
```
But perhaps there are various edge cases that make this hard.

Reply:
@rhshadrach This looks like syntactic sugar to me. I would leave it out here and put it in a follow-up pull request once the overall mechanism is settled.

Reply:
yes this change has other implications so I would also suggest making a follow on discussion.

Reply:
@moi90 - I would not describe this as syntactic sugar, but rather as correcting a defect. That said, it doesn't seem significant to me, and I agree with your approach, but would like others thoughts here.

cc @jreback, @simonjayhawkins

Reply:
For context: the current Styler has a method `maybe_convert_css_to_tuples` which is based on the `att: val; att2: val2;` format. It currently raises an error to user if format structure is not correct. If we deviate from this for LaTeX parsing, it would probably require changing the behaviour of that method, and would represent a departure from the make _'LaTeX Styler fit the normal CSS structure'_ and, in that case, it may well be worth  asking further questions about the best way to approach this, but my preference is to keep as is.

Reply:
> If we deviate from this for LaTeX parsing, it would probably require changing the behaviour of that method

I don't understand why we can't introduce a compatibility layer instead as I've suggested above. If you've said why that is not a viable route, I seem to have missed it.

Reply:
because 'props' is an argument to every built-in styling method (in this case `highlight_max`), which are all ignorant of whether you are styling CSS or LaTeX. So, if you add that compatibility layer to each one of these methods, i.e. **before** 'props' are all passed to `maybe_convert_css_to_tuple` you may as well add this compatability adjustment to `maybe_convert_css_to_tuples`, which will change the operation of that method but the code will, at least, have the same effect, not be duplicated, and not pollute all the built-in methods.

For transparency, I think `maybe_convert_css_to_tuples` happens late enough in the render process that a `latex` flag could differentiate between different rendering cases, and therefore introduce your code, without impacting CSS warnings.

I just preferred minimising additional complexity and relying on the user to fit the documented structure.

Reply:
Ah, I didn't realize there wasn't a better place to intercept. Agreed, and thanks for the response!